napari/layers/labels/labels.py

```diff
@@ -474,7 +474,16 @@ def color(self, color):
         }
         self._color = colors
 
-        # if colors just contains default colors for None and background
+        """
+        `colors` may contain just the default None and background label
+        colors, in which case we need to be in AUTO color mode. Otherwise,
+        `colors` contains colors for all labels, and we should be in DIRECT
+        mode.
+
+        For more information
+        - https://github.com/napari/napari/issues/2479
+        - https://github.com/napari/napari/issues/2953
+        """
```

Comment:
```suggestion
        # `colors` may contain just the default None and background label
        # colors, in which case we need to be in AUTO color mode. Otherwise,
        # `colors` contains colors for all labels, and we should be in DIRECT
        # mode.

        # For more information
        # - https://github.com/napari/napari/issues/2479
        # - https://github.com/napari/napari/issues/2953
```